src/index.ts

```diff
@@ -6,15 +6,17 @@ type ToupleArray = ReadonlyArray<any> | readonly [any];
 interface WrapRequestOptions<Y, T> {
     deps?: Y;
     defaultData?: T;
+    cacheKey?: string;
 }
 
 export function useWrapRequest<T, Y extends ToupleArray>(
     req: (...deps: Y) => Promise<T>,
     options: WrapRequestOptions<Y, T> = {}
 ) {
-    const [, setResult] = React.useState();
-    const [, setState] = React.useState();
-    const deps = (options.deps || []) as Y;
+    const [, setResult] = React.useState<T>();
+    const [, setState] = React.useState<string | Error>();
+    const {deps: oDepts, ...rest } = options;
```

Comment:
Can we be a little more specific in the naming? like `wrapRequestOptions` instead of just `rest`.